core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -995,6 +978,44 @@ private[spark] class AppStatusListener(
     }
   }
 
+  private def updateBroadcastBlock(
+      event: SparkListenerBlockUpdated,
+      broadcast: BroadcastBlockId): Unit = {
+    val now = System.nanoTime()
+    val executorId = event.blockUpdatedInfo.blockManagerId.executorId
+    val storageLevel = event.blockUpdatedInfo.storageLevel
+
+    // Whether values are being added to or removed from the existing accounting.
+    val diskDelta = event.blockUpdatedInfo.diskSize * (if (storageLevel.useDisk) 1 else -1)
+    val memoryDelta = event.blockUpdatedInfo.memSize * (if (storageLevel.useMemory) 1 else -1)
+
+    val maybeExec = liveExecutors.get(executorId)
+    updateExecutorMemoryDiskInfo(maybeExec, storageLevel, memoryDelta, diskDelta, None, None)
+    maybeExec.foreach(exec => maybeUpdate(exec, now))
+
+  }
+    
+  // update executor memory and disk usage info
+  private def updateExecutorMemoryDiskInfo(
+      maybeExec: Option[LiveExecutor],
+      storageLevel: StorageLevel,
+      memoryDelta: Long,
+      diskDelta: Long,
+      OffHeapDelta: Option[Long],
+      OnHeapDelta: Option[Long]): Unit = {
+    maybeExec.foreach { exec =>
```

Comment:
Move the `foreach` to the caller. That avoids repeating the `foreach`, and you could wrap more logic that doesn't need to run when the executor is not found.